cpp/oneapi/dal/algo/svm/common.hpp

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/data/table.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::svm {
+
+namespace detail {
+struct tag {};
+class descriptor_impl;
+class model_impl;
+} // namespace detail
+
+namespace method {
+struct thunder {};
+struct smo {};
+using by_default = thunder;
+} // namespace method
+
+namespace task {
+struct classification {};
+struct regression {};
+using by_default = classification;
+} // namespace task
+
+class descriptor_base : public base {
+public:
+    using tag_t    = detail::tag;
+    using float_t  = float;
+    using task_t   = task::by_default;
+    using method_t = method::by_default;
+
+    descriptor_base();
+
+    double get_c() const;
+    double get_accuracy_threshold() const;
+    std::int64_t get_max_iteration_count() const;
+    double get_cache_size() const;
+    double get_tau() const;
+    bool get_shrinking() const;
```

Comment:
Propably better:

```suggestion
    bool do_shrinking() const;
```

Reply:
Will it be very inconsistent with other methods of library parameters? This name of the parameter was in the old interface. This will also need to be discussed.

Reply:
In PCA we have the same situation, `isDeterministic` in old API and `get/set_deterministic` in the new. I'd keep `get/set_shrinking`